tools/testing/selftests/seccomp/seccomp_bpf.c

```diff
@@ -4060,6 +4070,27 @@ TEST(user_notification_addfd)
 	ASSERT_EQ(ioctl(listener, SECCOMP_IOCTL_NOTIF_RECV, &req), 0);
 	ASSERT_EQ(addfd.id, req.id);
 
+	/* Verify we can do an atomic addfd and send */
+	addfd.newfd = 0;
+	addfd.flags = SECCOMP_ADDFD_FLAG_SEND;
+	fd = ioctl(listener, SECCOMP_IOCTL_NOTIF_ADDFD, &addfd);
+	EXPECT_EQ(filecmp(getpid(), pid, memfd, fd), 0);
```

Comment:
It would be good to test that the fd was installed at the correct place in the fdtable (`addfd.newfd = 0;`):
```diff
-	EXPECT_EQ(filecmp(getpid(), pid, memfd, fd), 0);
+	EXPECT_EQ(fd, 0);
+	EXPECT_EQ(filecmp(getpid(), pid, memfd, 0), 0);
```

Reply:
That won't work, this won't be fd 0. To specify a remote fd number to use, you need to set `SECCOMP_ADDFD_FLAG_SETFD`. This is another flag and that setting this flag only does what we want is tested a few lines above this: https://github.com/kinvolk/linux/pull/6/files#diff-8933b3a4375dff05fe0d6851a10c22a2f291f9edaa8592c229d73f861eaf73b8L4035-L4040

Reply:
oh, I see... then, could you add a similar test on the correct value of fd? I see this piece of code a bit earlier in this file:

```
	/*
	 * The child has fds 0(stdin), 1(stdout), 2(stderr), 3(memfd),
	 * 4(listener), so the newly allocated fd should be 5.
	 */
	EXPECT_EQ(fd, 5);
```

Reply:
Oh, sure, added that!